static/js/src/advantage/contracts-api.js

```diff
@@ -306,3 +306,82 @@ export async function setAutoRenewal(value) {
   let data = await response.json();
   return data;
 }
+
+export async function postGuestFreeTrial({
+  email,
+  account_name,
+  name,
+  address,
+  productListingId,
+  quantity,
+}) {
+  const queryString = window.location.search; // Pass arguments to the flask backend eg. "test=backend=true"
+
+  let response = await fetch(`/advantage/post_guest_trial${queryString}`, {
+    method: "POST",
+    cache: "no-store",
+    credentials: "include",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      email: email,
+      account_name: account_name,
+      name: name,
+      address: address,
+      products: [
+        {
+          product_listing_id: productListingId,
+          quantity: quantity,
+        },
+      ],
+    }),
+  });
+
+  if (response.ok) {
+    const data = await response.json();
+    return { ok: true, ...data };
+  } else {
+    console.error(`${response.status} - ${response.statusText}`);
+    return { ok: false };
```

Comment:
When you can you want to keep the type and structure of a function return value the same so that consumers of the API don't have to be quite as defensive. 

In this case that would mean that both return objects would contain a data object, even if it's empty.

Reply:
The free trial endpoints (which are not used yet) were results of a prior iteration. I'll bring them in line with the rest of the calls :+1: